src/gate/library/liberty/net_data.cpp

```diff
@@ -0,0 +1,61 @@
+//===----------------------------------------------------------------------===//
+//
+// Part of the Utopia EDA Project, under the Apache License v2.0
+// SPDX-License-Identifier: Apache-2.0
+// Copyright 2023 ISP RAS (http://www.ispras.ru)
+//
+//===----------------------------------------------------------------------===//
+
+#include "gate/library/liberty/net_data.h"
+#include "gate/simulator/simulator.h"
+
+#include <unordered_map>
+
+namespace GModel = eda::gate::model;
+
+using eda::gate::model::Gate;
+using eda::gate::simulator::Simulator;
+using eda::gate::optimizer::RWDatabase;
+
+void NetData::fillDatabase(RWDatabase &database) {
+  RWDatabase::InputId id = 0;
+  std::unordered_map<RWDatabase::TruthTable, RWDatabase::BoundGNetList> storage;
+  for (auto &net: combNets) {
+    RWDatabase::BoundGNet bounder;
+    for (auto link: net->sourceLinks()) {
+      bounder.bindings.emplace(id++, link.target);
+    }
+    RWDatabase::TruthTable key = NetData::buildTruthTab(net.get())[0];
+    bounder.net.reset(net.release());
+    storage[key].push_back(bounder);
+  }
```

Comment:
Inputs are enumerated per cell, so reset the counter for every cell.

```suggestion
  std::unordered_map<RWDatabase::TruthTable, RWDatabase::BoundGNetList> storage;
  for (auto &net: combNets) {
    RWDatabase::InputId id = 0;
    RWDatabase::BoundGNet bounder;
    for (auto link: net->sourceLinks()) {
      bounder.bindings.emplace(id++, link.target);
    }
    RWDatabase::TruthTable key = NetData::buildTruthTab(net.get())[0];
    bounder.net.reset(net.release());
    storage[key].push_back(bounder);
  }
```